src/java/org/apache/cassandra/tools/nodetool/CommandUtils.java

```diff
@@ -0,0 +1,239 @@
+    }
+
+    public static List<String> concatArgs(String first, String second, String[] third)
+    {
+        return concat(ofNullable(first), concatArgs(second, third).stream()).collect(Collectors.toList());
+    }
+
+    public static List<String> concatArgs(String first, String second, List<String> third)
+    {
+        return concat(ofNullable(first), concatArgs(second, third).stream()).collect(Collectors.toList());
+    }
+
+    public static List<String> concatArgs(String first, String second, String third, String[] fourth)
+    {
+        return concat(ofNullable(first), concatArgs(second, third, fourth).stream()).collect(Collectors.toList());
+    }
+
+    public static void printSet(PrintStream out, String colName, Set<String> values)
+    {
+        if (values == null || values.isEmpty())
+            return;
+
+        TableBuilder table = new TableBuilder();
+
+        table.add(colName + ": ");
+
+        for (String value : values)
+            table.add(value);
+
+        table.printTo(out);
+    }
+
+    public static List<String> parseOptionalKeyspaceAccordManaged(List<String> cmdArgs, NodeProbe nodeProbe)
+    {
+        return parseOptionalKeyspace(cmdArgs, nodeProbe, KeyspaceSet.ACCORD_MANAGED);
+    }
+
+    public static List<String> parseOptionalKeyspace(List<String> cmdArgs, NodeProbe nodeProbe)
+    {
+        return parseOptionalKeyspace(cmdArgs, nodeProbe, KeyspaceSet.ALL);
+    }
+
+    public static List<String> parseOptionalKeyspaceNonLocal(List<String> cmdArgs, NodeProbe nodeProbe)
+    {
+        return parseOptionalKeyspace(cmdArgs, nodeProbe, KeyspaceSet.NON_LOCAL_STRATEGY);
+    }
+
+    private static List<String> parseOptionalKeyspace(List<String> cmdArgs, NodeProbe nodeProbe, KeyspaceSet defaultKeyspaceSet)
+    {
+        List<String> keyspaces = new ArrayList<>();
+
+
+        if (cmdArgs == null || cmdArgs.isEmpty())
+        {
+            if (defaultKeyspaceSet == KeyspaceSet.NON_LOCAL_STRATEGY)
+                keyspaces.addAll(keyspaces = nodeProbe.getNonLocalStrategyKeyspaces());
+            else if (defaultKeyspaceSet == KeyspaceSet.NON_SYSTEM)
+                keyspaces.addAll(keyspaces = nodeProbe.getNonSystemKeyspaces());
+            else if (defaultKeyspaceSet == KeyspaceSet.ACCORD_MANAGED)
+                keyspaces.addAll(nodeProbe.getAccordManagedKeyspaces());
+
+            else
+                keyspaces.addAll(nodeProbe.getKeyspaces());
+        }
+        else
+        {
+            keyspaces.add(cmdArgs.get(0));
+        }
+
+        for (String keyspace : keyspaces)
+        {
+            if (!nodeProbe.getKeyspaces().contains(keyspace))
+                throw new IllegalArgumentException("Keyspace [" + keyspace + "] does not exist.");
+        }
+
+        return Collections.unmodifiableList(keyspaces);
+    }
+
+    public static String[] parseOptionalTables(List<String> cmdArgs)
```

Comment:
probably a comment/javadoc is needed about an input/output format here, not obvious..